src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
I marked "Can't repurpose preexisting or third party attributes" as a pro and con because I think there is room for debate on whether that is strictly a good thing, especially with well known attributes of which people already expect specific meanings and interpretations.

I'm not suggesting that all of the pros/cons should be weighted equally, so feel free to weigh each as you feel appropriate (including 0 weight!).